### PR TITLE
task_6623_init_api_fetch_data_collections

### PR DESCRIPTION
**Pull request by QNFC**
task_6623_init_api_fetch_data_collections
Link redmine: https://edu-redmine.framgia.vn/issues/6623

### CODE REVIEW

**Comment by QNFC** [comment]
ready

**Comment by tuanlvt-1117** [comment]
thiếu link redmine

**Comment by tuanlvt-1117** [comment]
commented

**Comment by QNFC** [comment]
> commented

e quen :)

**Comment by QNFC** [comment]
ready

> commented

**Comment by QNFC** [comment]
ready

**Comment by QNFC** [comment]
ready

File: app/src/main/java/com/framgia/my_editor_02/data/repository/ImageRepository.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import com.framgia.my_editor_02.data.source.remote.ImageRemoteDataSource;
+import io.reactivex.Single;
+import java.util.List;
+
+public class ImageRepository {
+    private static ImageRepository sInstance;
+    private ImageRemoteDataSource mRemoteDataSource;
+    private ImageLocalDataSource mLocalDataSource;
+
+    public ImageRepository(ImageRemoteDataSource remoteDataSource,
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/data/repository/ImageRepository.java]: check access modifier

File: app/src/main/java/com/framgia/my_editor_02/data/source/remote/ImageRemoteDataSource.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import com.framgia.my_editor_02.data.source.remote.config.service.ImageApi;
+import com.framgia.my_editor_02.data.source.remote.config.service.ServiceClient;
+import io.reactivex.Single;
+import java.util.List;
+
+public class ImageRemoteDataSource implements ImageDataSource.ImageRemoteDataSource {
+    private static ImageRemoteDataSource sInstance;
+    private ImageApi mImageApi;
+
+    public ImageRemoteDataSource(ImageApi imageApi) {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/data/source/remote/ImageRemoteDataSource.java]: check access modifier

File: app/src/main/java/com/framgia/my_editor_02/data/source/remote/config/service/ServiceClient.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.google.gson.Gson;
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class ServiceClient {
+    private static final String BASE_API_URL ="https://api.unsplash.com/";
+    public static ImageApi createServiceClient() {
+        Retrofit mRetrofit = new Retrofit.Builder().baseUrl(BASE_API_URL)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/data/source/remote/config/service/ServiceClient.java]:
```suggestion
        Retrofit retrofit = new Retrofit.Builder().baseUrl(BASE_API_URL)
```

File: app/src/main/java/com/framgia/my_editor_02/data/source/remote/config/service/ServiceClient.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+                .addConverterFactory(GsonConverterFactory.create(new Gson()))
+                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+                .build();
+        return mRetrofit.create(ImageApi.class);
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/data/source/remote/config/service/ServiceClient.java]: return luôn nguyên 1 đùm trên luôn e

File: app/src/main/java/com/framgia/my_editor_02/screen/collections/CollectionsFragment.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import com.framgia.my_editor_02.databinding.FragmentCollectionsBinding;
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+public class CollectionsFragment extends Fragment {
+    private static CollectionsFragment sInstance;
+    private FragmentCollectionsBinding mBinding;
+    private CollectionsViewModel mViewModel;
+    private ImageRepository mRepository;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/screen/collections/CollectionsFragment.java]: 21 - 23
convert to local

File: app/src/main/java/com/framgia/my_editor_02/screen/collections/CollectionsFragment.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import com.framgia.my_editor_02.data.repository.ImageRepository;
+import com.framgia.my_editor_02.data.source.Local.ImageLocalDataSource;
+import com.framgia.my_editor_02.data.source.remote.ImageRemoteDataSource;
+import com.framgia.my_editor_02.databinding.FragmentCollectionsBinding;
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+public class CollectionsFragment extends Fragment {
+    private static CollectionsFragment sInstance;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/screen/collections/CollectionsFragment.java]: no used

**Comment by QNFC** [comment]
ready

File: app/src/main/java/com/framgia/my_editor_02/screen/collections/CollectionsFragment.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+import com.framgia.my_editor_02.data.repository.ImageRepository;
+import com.framgia.my_editor_02.data.source.Local.ImageLocalDataSource;
+import com.framgia.my_editor_02.data.source.remote.ImageRemoteDataSource;
+import com.framgia.my_editor_02.databinding.FragmentCollectionsBinding;
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+public class CollectionsFragment extends Fragment {
+    private int DEFAULT_PAGE = 1;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/screen/collections/CollectionsFragment.java]: check convention

File: app/src/main/java/com/framgia/my_editor_02/screen/home/HomeFragment.java [hunk 8]
```diff
@@ -22,6 +23,7 @@
  */
 public class HomeFragment extends Fragment implements TabLayout.OnTabSelectedListener {
     public static final String TAG = HomeFragment.class.getSimpleName();
+    public static final String COLLECTIONS = "COLLECTIONS";
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/my_editor_02/screen/home/HomeFragment.java]: check access modifier

**Comment by QNFC** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**